### PR TITLE
Gpu cherry picks

### PR DESCRIPTION
**Pull request by Klaven**
These are cherry picks from what we did for 4.5 as we will start supporting 4.2 in CaaSP in the next few weeks. 

Should not get merged until we do another 4.2 release. I think 4.2.3. Anyhow. 

I made this to the `4.2.0 branch as I did not see a 4.2.2 branch. If there is a better place to put these please let me know and I will do that.

# Describe your changes
What is this PR about? Please briefly detail why this needs to be integrated
and where you think it should go if it's not obvious from the changes.

# Related Issues / Projects
Please provide links to Bugzilla and other GitHub projects with your description if they are related to the changes.

Relates to:

### CODE REVIEW

**Review by cmurphy — CHANGES_REQUESTED** [review]
There seem to be a lot of changes unrelated to the topic "Gpu cherry picks", are these necessary for this PR? Could they be split out?

File: adoc/admin-gpus.adoc [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+[source,bash]
+----
+$ kubectl taint nodes worker0 nvidia.com/gpu=:NoSchedule
+----
+
+See the Kubernetes documentation on link:{kubedoc}concepts/scheduling-eviction/taint-and-toleration/[taints and tolerations] for a discussion on the considerations for using the NoSchedule or PreferNoSchedule effects. If you use the NoSchedule effect, you must also add the appropriate toleration to infrastructure-critical Daemonsets that must run on all nodes, such as the kured, kube-proxy, and cilium Daemonsets.
+
+[NOTE]
+====
+The link:{kubedoc}reference/access-authn-authz/admission-controllers/#extendedresourcetoleration[ExtendedResourceToleration admission controller] is enabled on {productname} v5 by default. This is a mutating admission controller that reviews all pod requests and adds tolerations to any pod that requests an extended resource advertised by a device plugin. For the NVIDIA GPU device plugin, it will automatically add the nvidia.com/gpu toleration to pods that request the nvidia.com/gpu resource, so you will not need to add this toleration explicitly for every GPU workload.
```
[the remark this answers is not in the document]
Comment (Klaven) [reply adoc/admin-gpus.adoc:103]: @cmurphy I was going to write something about taints and tolerations today. Not a full overview but just a quick (make sure to remember) Thanks for looking over this and catching these things!

**Comment by Klaven** [comment]
rebased too the right branch. will make a new commit with my updates too the documentation to reflect the differences in v4.2

File: adoc/admin-gpus.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (cmurphy) [reply adoc/admin-gpus.adoc:103]: I don't really think anything needs to be written? Just 363075d needs to be removed.

**Comment by cmurphy** [comment]
It seems like instead of removing the commit I was talking about it got squashed into dcd2a8c0369765e4773315dc39361e4bfe199f02 ? and then undone in 7803b050641d34d91c3defa1c41a52971af3ece9 ? So it's not a true "cherry pick" series but the content looks good to me now :+1:

**Comment by Klaven** [comment]
@nkoranova do you mind if the history is not a "pure cherry pick"? if it's not a big deal we can merge this. Just let me know and I'll redo it if you want me too.

**Comment by Klaven** [comment]
@r0ckarong same question too you?

**Review by tdaines42 — COMMENTED** [review]
The content lgtm. I'm guessing since this is a backport that the formatting issues the CI is calling out is ok? As long as the Docs team is good with it I am

**Comment by cmurphy** [comment]
I went ahead and fixed the cherry-picks since it bothered me.

**Review by Klaven — COMMENTED** [review]
@cmurphy cleaned up the history as it should be, thank you. This looks good now.

**Comment by r0ckarong** [comment]
@davidcassany Is this going to be part of the 4.2.4 release?

**Comment by r0ckarong** [comment]
CaaSP is in maintenance and only hi-prio security fixes are going to be added.

**Pull request closed without merging**